OpenSim/Common/C3DFileAdapter.cpp

```diff
@@ -108,13 +112,21 @@ C3DFileAdapter::extendRead(const std::string& fileName) const {
 
         double time_step{1.0 / acquisition->GetPointFrequency()};
         for(int f = 0; f < marker_nrow; ++f) {
-            SimTK::RowVector_<SimTK::Vec3> row{marker_pts->GetItemNumber()};
+            SimTK::RowVector_<SimTK::Vec3> row{ marker_pts->GetItemNumber(), 
+                                                SimTK::Vec3(SimTK::NaN) };
             int m{0};
             for(auto it = marker_pts->Begin();  it != marker_pts->End(); ++it) {
                 auto pt = *it;
-                row[m++] = SimTK::Vec3{pt->GetValues().coeff(f, 0),
-                                       pt->GetValues().coeff(f, 1),
-                                       pt->GetValues().coeff(f, 2)};
+                // BTK reads empty values as zero, but sets a "residual" value
+                // to -1 and it is how it knows to export these values as 
+                // blank, instead of 0,  when exporting to .trc
+                if (!pt->GetValues().row(f).isZero() &&    //not precisely zero
```

Comment:
Why is it necessary to include `!pt->GetValues().row(f).isZero()` in the condition? Shouldn't the NaNs (initialized at L115–116) be overwritten with zeros if the zeros are real data (i.e., if "residual" is not -1)? It looks like (0,0,0) becomes NaNs in `row` regardless of the value of the "residual" flag, which seems to contradict the comment.